src/nm-ip4-config.c

```diff
@@ -2933,13 +2933,21 @@ nm_ip4_config_hash (const NMIP4Config *self, GChecksum *sum, gboolean dns_only)
 	}
 
 	val = nm_ip4_config_mdns_get (self);
-	g_checksum_update (sum, (const guint8 *) &val, sizeof (val));
+	if (val != NM_SETTING_CONNECTION_MDNS_DEFAULT)
+		g_checksum_update (sum, (const guint8 *) &val, sizeof (val));
 
 	val = nm_ip4_config_llmnr_get (self);
-	g_checksum_update (sum, (const guint8 *) &val, sizeof (val));
-
-	val = nm_ip4_config_get_dns_priority (self);
-	g_checksum_update (sum, (const guint8 *) &val, sizeof (val));
+	if (val != NM_SETTING_CONNECTION_LLMNR_DEFAULT)
+		g_checksum_update (sum, (const guint8 *) &val, sizeof (val));
+
+	/* FIXME: the DNS priority should be considered as well,
```

Comment:
Where is this assumption made? Can you at the same time also add a FIXME comment at the place where this assumption is made?

Recently, I started grouping FIXME comments like "FIXME(shutdown)" and "FIXME(copy-on-write-connection)". Can you group these two FIXME statements together?


Btw, the approach of computing the MD5 sum of NMIPXConfig instance for comparing them is significantly flawed. We should implement proper hash/equals/cmp functions, like we commonly do. That warrants a FIXME comment in extra big letters.


Btw, in case you didn't know, instead of "fixup! " in the commit message, you can use "squash!". Also, `git commit --squash=$COMMIT`.

Reply:
Ok, I've added a FIXME and improved the existing one.